internal/zinc-core/src/main/java/xsbti/compile/AnalysisStore.java

```diff
@@ -0,0 +1,37 @@
+package xsbti.compile;
+
+import java.util.Optional;
+
+/**
+ * Defines a store interface that provides analysis read and write capabilities to users.
+ *
+ * The store is a backend-independent interface that allows implementors to decide how
+ * the analysis stores are read and written before or after every incremental compile.
+ */
+public interface AnalysisStore {
```

Comment:
I can split it. The reason why this is in the same PR is because the protobuf PR touches all these utilities. Breaking it in another PR is going to force me to depend on this, rebase until it's done, and then create an extra PR in sbt/sbt to synchronize with this. Is this worth it?